include/Core/Binary_Init_File.hpp

```diff
@@ -14,28 +15,49 @@ namespace fug {
 
     FUG_RESOURCE_INITINFO_INIT(Binary, BinaryInitInfo_File)
     {
+        // Check that keys are valid
+        checkJsonValidity("BinaryInitInfo_File", json, {"type", "fileName"}, {"fileName"});
+
+        // Parse fields
         std::string fileName = json["fileName"];
         initInfo.fileName = std::string(FUG_RES_DIRECTORY) + fileName;
     }
 
     FUG_RESOURCE_INIT(Binary, BinaryInitInfo_File) {
+        // Open file
         FILE *fp = fopen(initInfo.fileName.c_str(), "rb");
         if (!fp) {
-            throw "Binary: open failed";
+            FUG_LOG(LogLevel::Error)("BinaryInitInfo_File: fopen failed\n");
+            FUG_LOG(LogLevel::Error)("File: %s\n", initInfo.fileName.c_str());
+            throw;
         }
-        if (fseek(fp, 0, SEEK_END) < 0) {
+        // Get size of file
+        int err = 0;
+        if ((err = fseek(fp, 0, SEEK_END)) < 0) {
             fclose(fp);
-            throw "Binary: fseek failed";
+            FUG_LOG(LogLevel::Error)("BinaryInitInfo_File: fseek SEEK_END failed\n");
+            FUG_LOG(LogLevel::Error)("File: %s\n", initInfo.fileName.c_str());
+            FUG_LOG(LogLevel::Error)("Error: %i\n", err);
```

Comment:
This print does not have any value in itself as err is always going to be -1 if this branch is taken. I suggest either removing this or printing strerror(errno).